lib/streamlit/server/routes.py

```diff
@@ -63,10 +63,22 @@ def set_extra_headers(self, path):
     def parse_url_path(self, url_path: str) -> str:
         url_parts = url_path.split("/")
 
-        # TODO(vdonato): Maybe clean this up if we decide to tweak the spec so
-        # that there's a separator between the baseUrlPath and page name.
         maybe_page_name = url_parts[0]
         if maybe_page_name in self._pages:
+            # If we're trying to navigate to a page, we return "index.html"
```

Comment:
Should we say `If we're trying to navigate to the main page to avoid ambiguity in the comment?

Reply:
This returns "index.html" to short-circuit the redirect+extra roundtrip for any page, not just the main one, so I'll leave the comment as-is